File: crypto-casino-backend/src/main/java/hu/bme/aut/crypto_casino_backend/dto/stats/UserStatsResponse.java
package hu.bme.aut.crypto_casino_backend.dto.stats;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.Map;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserStatsResponse {

  private Integer totalGamesPlayed;

  private Double winRate;

  private BigDecimal totalWinnings;

  private BigDecimal totalLosses;

  private BigDecimal netProfitLoss;

  private BigDecimal biggestWin;

  private String mostPlayedGame;

  private BigDecimal totalDeposited;

  private BigDecimal totalWithdrawn;

  private BigDecimal totalWagered;

  private BigDecimal houseEdgePaid;

  private Map<String, GameStatsDto> gameStats;

}

File: crypto_casino_android/app/src/main/java/hu/bme/aut/crypto_casino_android/data/model/stats/UserStatsResponse.kt
package hu.bme.aut.crypto_casino_android.data.model.stats

import java.math.BigDecimal

data class UserStatsResponse(
    val totalGamesPlayed: Int,
    val winRate: Double,
    val totalWinnings: BigDecimal,
    val totalLosses: BigDecimal,
    val netProfitLoss: BigDecimal,
    val biggestWin: BigDecimal,
    val mostPlayedGame: String?,
    val totalDeposited: BigDecimal,
    val totalWithdrawn: BigDecimal,
    val totalWagered: BigDecimal,
    val houseEdgePaid: BigDecimal,
    val gameStats: Map<String, GameStatsDto>
)
